File: sdks/go/pkg/models/operations/putfoldersidwatermark.go
package operations

import (
	"openapi/pkg/models/shared"
	"openapi/pkg/utils"
)

type PutFoldersIDWatermarkPathParams struct {
	FolderID string `pathParam:"style=simple,explode=false,name=folder_id"`
}

type PutFoldersIDWatermarkRequestBodyWatermarkImprintEnum string

const (
	PutFoldersIDWatermarkRequestBodyWatermarkImprintEnumDefault PutFoldersIDWatermarkRequestBodyWatermarkImprintEnum = "default"
)

// PutFoldersIDWatermarkRequestBodyWatermark
// The watermark to imprint on the folder
type PutFoldersIDWatermarkRequestBodyWatermark struct {
	Imprint PutFoldersIDWatermarkRequestBodyWatermarkImprintEnum `json:"imprint"`
}

type PutFoldersIDWatermarkRequestBody struct {
	Watermark PutFoldersIDWatermarkRequestBodyWatermark `json:"watermark"`
}

type PutFoldersIDWatermarkRequest struct {
	Retries    *utils.RetryConfig
	PathParams PutFoldersIDWatermarkPathParams
	Request    *PutFoldersIDWatermarkRequestBody `request:"mediaType=application/json"`
}

type PutFoldersIDWatermarkResponse struct {
	ClientError *shared.ClientError
	ContentType string
	StatusCode  int64
	Watermark   *shared.Watermark
}

File: sdks/ts/src/sdk/models/operations/putfoldersidwatermark.ts
import { SpeakeasyMetadata, SpeakeasyBase, RetryConfig } from "../../../internal/utils";
import * as shared from "../shared";



export class PutFoldersIdWatermarkPathParams extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "pathParam, style=simple;explode=false;name=folder_id" })
  folderId: string;
}

export enum PutFoldersIdWatermarkRequestBodyWatermarkImprintEnum {
    Default = "default"
}


// PutFoldersIdWatermarkRequestBodyWatermark
/** 
 * The watermark to imprint on the folder
**/
export class PutFoldersIdWatermarkRequestBodyWatermark extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=imprint" })
  imprint: PutFoldersIdWatermarkRequestBodyWatermarkImprintEnum;
}


export class PutFoldersIdWatermarkRequestBody extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=watermark" })
  watermark: PutFoldersIdWatermarkRequestBodyWatermark;
}


export class PutFoldersIdWatermarkRequest extends SpeakeasyBase {
  @SpeakeasyMetadata()
  retries?: RetryConfig;

  @SpeakeasyMetadata()
  pathParams: PutFoldersIdWatermarkPathParams;

  @SpeakeasyMetadata({ data: "request, media_type=application/json" })
  request?: PutFoldersIdWatermarkRequestBody;
}


export class PutFoldersIdWatermarkResponse extends SpeakeasyBase {
  @SpeakeasyMetadata()
  clientError?: shared.ClientError;

  @SpeakeasyMetadata()
  contentType: string;

  @SpeakeasyMetadata()
  statusCode: number;

  @SpeakeasyMetadata()
  watermark?: shared.Watermark;
}
